starknet_py/tests/e2e/mock/compile_contracts.sh

```diff
@@ -1,32 +1,85 @@
 #!/bin/bash
-
+set -e
 MOCK_DIRECTORY=starknet_py/tests/e2e/mock
-CONTRACTS_DIRECTORY="$MOCK_DIRECTORY"/contracts
-CONTRACTS_COMPILED_DIRECTORY="$MOCK_DIRECTORY"/contracts_compiled
-
-# delete all artifacts except precompiled ones
-find $CONTRACTS_COMPILED_DIRECTORY -maxdepth 1 -type f -delete
+CONTRACTS_DIRECTORY_V1="$MOCK_DIRECTORY"/contracts_v1
+CONTRACTS_DIRECTORY_V2="$MOCK_DIRECTORY"/contracts_v2
 
-# compile Cairo test contracts
-echo "Compiling Cairo contracts with $(poetry run starknet-compile-deprecated --version)"
+setup_scarb() {
+    SCARB_VERSION=$1
```

Comment:
```suggestion
    SCARB_VERSION="$1"
```

Please double quote variables to prevent globbing or splitting.
This comment applies to all variables you use in this script